### PR TITLE
Added hash algorithm to the config as an option to set at runtime

### PR DESCRIPTION
**Pull request by hubchub**

### CODE REVIEW

**Comment by hubchub** [comment]
Depends on https://github.com/dominant-strategies/go-quai/pull/1025

Don't merge until above does and I will have to update go.mod before merging

File: config/config.yaml.dist
Comment (Djadih) [inline config/config.yaml.dist:12]: Personally I feel hesitant about including this flag, just because it's going to be totally deprecated. But we also distribute it in go-quai.
@wizeguyy what do you think?

**Review by Djadih — COMMENTED** [review]
Overall this is awesome. Not sure about distributing the flag.

File: config/config.yaml.dist
Comment (hubchub) [reply config/config.yaml.dist:12]: it sucks switching between branches constantly

I could use worktrees I guess but couldn't we just add an issue to remove later?

File: go.mod
Comment (hubchub) [inline go.mod]: this will be dropped when go-quai@1025 is merged

File: config/config.yaml.dist
Comment (Djadih) [reply config/config.yaml.dist:12]: I was only referring to having the flag in the dist or not. I agree this is a much needed change.

These are the three scenarios:
- Copy the dist into network.env and run: runs progpow
- Add runblake3: false: runs progpow
- Add runblake3: true: runs blake3

Comment (Djadih) [reply config/config.yaml.dist:12]: Discussed offline

File: util/config.go
Comment (wizeguyy) [inline util/config.go]: (nit) indentation

**Review by wizeguyy — COMMENTED** [review]
love it. Thanks for putting this up! Other than a small formatting comment, my only issue is the PR name :)

Once you fix those, its good to merge, IMO

**Review by wizeguyy — APPROVED** [review]

**Pull request merged**
